### ISSUE TITLE
Guzzle 6

### ISSUE THREAD

**Issue by CDyWeb** [commerceguys/guzzle-oauth2-plugin#19]
Please make this library compatible with Guzzle version 6

**Comment by ip512** [comment]
:+1:

**Comment by djfurman** [comment]
:+1:

**Comment by santiph** [comment]
:+1:

**Comment by spinx** [comment]
:+1:

**Comment by nicekiwi** [comment]
:+1:

**Comment by ldiebold** [comment]
:+1:

**Comment by obarcelonap** [comment]
:+1:

**Comment by frankkessler** [comment]
I have the code working here with Guzzle 6, but I haven't had time to update the unit tests and really clean it up:  https://github.com/frankkessler/guzzle-oauth2-plugin

**Comment by ghazanfarmir** [comment]
I still get this

```
commerceguys/guzzle-oauth2-plugin v2.1.0 requires guzzlehttp/guzzle ~5.0 -> no matching package found 
```
with following composer configuration:
```
"require": {
    "guzzlehttp/guzzle": "^6.1",
    "commerceguys/guzzle-oauth2-plugin": "~2.0"
  }
```

**Comment by RomeroMsk** [comment]
:+1:

**Comment by fabpot** [comment]
ahah, first time I'm going to do something I hate to see people doing on my repositories, sorry guys, :+1:

**Comment by pjcdawkins** [comment]
The direction my thoughts are going in is to remove almost all the code dealing with OAuth2... and just provide a middleware, to link this existing oauth2 library with Guzzle 6+:

https://github.com/thephpleague/oauth2-client

**Comment by liverbool** [comment]
:+1:

**Comment by vzailskas** [comment]
:+1:

**Comment by lemasson-h** [comment]
I created a pull request to have it working with guzzle 6: https://github.com/commerceguys/guzzle-oauth2-plugin/pull/29. If you could let me know what you think about it, that would be great. :)

**Comment by mathielen** [comment]
:+1:

**Comment by geekdevs** [comment]
@pjcdawkins I agree what you say about switching this to middleware. I even found some for the good start (https://github.com/gsomoza/oauth2-middleware).

The problem is that it basically means implementations would need to be re-created to support Guzzle 6 then. Currently I some library I'd like to use which uses Guzzle 6, but cannot because this `guzzle-oauth2-plugin` is Guzzle 5 and thus conflicts with v.6.

So I am looking forward to get that PR merged! I think I'm not alone in that issue.

**Comment by pau1m** [comment]
👍

**Comment by guiyomh** [comment]
:+1:

**Comment by zazoomauro** [comment]
👍

**Comment by august-9426** [comment]
+ 1

**Comment by ghost** [comment]
tested working like charm and can not be easier https://github.com/dunnleaddress/oauth2-client
